servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1031,44 +1042,43 @@ private void journalCreatePathResult(InodeTree.CreatePathResult createResult) {
   /**
    * Renames a file to a destination. Called via RPC.
    *
-   * @param fileId the source file to rename
+   * @param srcPath the source path to rename
    * @param dstPath the destination path to rename the file to
-   * @return true if the operation was successful and false otherwise
    * @throws FileDoesNotExistException if a non-existent file is encountered
    * @throws InvalidPathException if an invalid path is encountered
    * @throws IOException if an I/O error occurs
    */
-  public boolean rename(long fileId, TachyonURI dstPath)
-      throws FileDoesNotExistException, InvalidPathException, IOException {
+  public void rename(TachyonURI srcPath, TachyonURI dstPath) throws FileAlreadyExistsException,
+      FileDoesNotExistException, InvalidPathException, IOException {
     MasterContext.getMasterSource().incRenamePathOps(1);
     synchronized (mInodeTree) {
-      Inode srcInode = mInodeTree.getInodeById(fileId);
-      TachyonURI srcPath = mInodeTree.getPath(srcInode);
+      Inode srcInode = mInodeTree.getInodeByPath(srcPath);
       // Renaming path to itself is a no-op.
       if (srcPath.equals(dstPath)) {
-        return true;
+        return;
       }
       // Renaming the root is not allowed.
       if (srcPath.isRoot() || dstPath.isRoot()) {
-        return false;
+        throw new InvalidPathException(ExceptionMessage.ROOT_CANNOT_BE_RENAMED.getMessage());
```

Comment:
root is only being renamed if it's the source, we should have a different message when renaming something to root